### ISSUE TITLE
bun doesn't work correctly with `ky` - a popular fetching library

### ISSUE THREAD

**Issue by blitss** [oven-sh/bun#11682]
[`ky` ](https://github.com/sindresorhus/ky) is a popular fetching library based on `fetch` which has over 10k stars on Github. It doesn't work nice with Bun and sometimes provides undefined instead of json output.

### What version of Bun is running?

1.1.12+43f0913c3

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

create repro.ts file and run it:

```typescript
import _ky from 'ky'

const ky = _ky.extend({
  "method": "GET",
})

while (true) {
  const fetched = await ky('https://jsonplaceholder.typicode.com/posts/1').json<any>()
  console.log(fetched.id)
}
```

### What is the expected behavior?

Bun expected to output 
```
1
1
1
1
1
1
1
1
```
Literally always one

### What do you see instead?

It randomly provides undefined values, which doesn't happen with node
```
$ bun repro.ts
undefined
undefined
undefined
1
undefined
1
1
undefined
undefined
undefined
1
1
1
1
1
undefined
1
undefined
1
1
1
1
1
1
```

**Comment by devunt** [comment]
Related to #1381, #2477, #6348. Possible fix in #12473 is waiting for landing.

**Comment by LucHermkens** [comment]
Fixed in [Bun v1.1.27](https://github.com/oven-sh/bun/releases/tag/bun-v1.1.27)

**Comment by blitss** [comment]
Issue is no longer reproducible with the new fix, closing this issue
